doc/source/plugin_development.rst

```diff
@@ -235,27 +235,38 @@ to return a class inheriting from ``BaseUIHooksManager``. This class has
 specific methods to be reimplemented to perform operations before and after
 some UI operations.
 
-Data views
-^^^^^^^^^^
-
-A plugin can also define one or more custom visualization classes, and make
-them discoverable by a GUI application (wfmanager) through the
-``get_data_views`` method of the plugin class::
-
-    def get_data_views(self):
-        """ Returns a list of classes, which must inherit from
-        `force_wfmanager.ui.review.data_view.BaseDataView` or from one
-        of its subclasses.
+Service Offers Plugin
+^^^^^^^^^^^^^^^^^^^^^
+
+A plugin can also define one or more custom visualization classes for the
+GUI application ``force-wfmanager``, typically to either display data or
+provide a tailor-made UI for a specific user. In which case, the plugin class
+must inherit from ``force_bdss.core_plugins.service_offers_plugin.ServiceOffersPlugin``
+, which is a child class of ``BaseExtensionPlugin``. Any UI subclasses
+can then be made discoverable by ``force-wfmanager`` using the ``envisage``
+``ServiceOffer`` protocol through the ``get_service_offers_factories`` method::
+
+    def get_service_offer_factories(self):
+        """A method returning a list user-made objects to be provided by this
+        plugin as envisage ServiceOffer objects. Each item in the outer list is
+        a tuple containing an Interface trait to be used as the ServiceOffer
+        protocol and an inner list of subclass factories to be instantiated
+        from said protocol.
+
+        Returns
+        -------
+        service_offer_factories: list of tuples
+            List of objects to load, where each tuple takes the form
+            (Interface, [HasTraits1, HasTraits2..]), defining a Traits
+            Interface subclass and a list of HasTraits subclasses to be
+            instantiated as an envisage ServiceOffer.
         """
 
-Note that this method doesn't exist in ``force_bdss.api.BaseExtensionPlugin``
-and there is no need to override it, unless you want to provide custom data
-views.
-
 Make sure to import the module containing the data view class from inside
-``get_data_views``: this ensures that running BDSS without a GUI application
-doesn't import the graphical stack. For instance::
+``get_service_offer_factories``: this ensures that running BDSS without a GUI
+application doesn't import the graphical stack. For instance::
 
-    def get_data_views(self):
+    def get_service_offer_factories(self):
+        from force_wfmanager.ui import IBasePlot
         from .example_data_views import ExampleCustomPlot
-        return [ExampleCustomPlot]
+        return [(IBasePlot, [ExampleCustomPlot])]
```

Comment:
I'm not sure I understand, are you refactoring things in such a way that the `data_views` also go through `service_offer`s?

Reply:
Yes, in order to make a generic way for developers to contribute UI components with their plugins - do you think this might cause any problems?

Reply:
I think it's a good idea but it might need some work across the repositories (to replace the `get_data_views` calls)